api_scripts/python/simple_input.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+import json
+
+# getpass and input are simple ways to get user input
+import getpass
+
+import requests
+import pandas as pd
+from builtins import input
+
+# Static settings
+# Using a base urls is useful for switching between test and production environments easily
+BASE_URL = 'https://canvas.ubc.ca'
+PER_PAGE = 100
+
+# User input settings
+# token should be treated as a password (not visible when typed)
+token = None
+while not token:
+    token = getpass.getpass('Enter your access token:')
+auth_header = {'Authorization': 'Bearer ' + token} # setup the authorization header to be used later
+
+# require that the sort order to be provided
+sort_by = None
+while not sort_by in ["course_name", "sis_course_id", "teacher", "account_name"]:
+    sort_by = input("Select a sort order [course_name, sis_course_id, teacher, account_name]:")
+
+
+print("Finding courses...")
+print("-----------------------------")
+# continue to make requests until all data has been received
+page = 1
+courses = []
+while True:
+    # request urls should always be based of the base url so they do not
+    # need to be changed when switching between test and production environments
+    request_url = BASE_URL + '/api/v1/courses'
+    params = {
+        "per_page": str(PER_PAGE),
+        "page": str(page),
+        "sort": sort_by,
```

Comment:
Does this actually work for you? I tested, and couldn't get sorting to work. It looks to me like the courses endpoint doesn't have a sort param:

https://canvas.instructure.com/doc/api/courses.html#method.courses.index